### PR TITLE
feat: ghost after-image

### PR DESCRIPTION
**Pull request by alicek0**
# Feature: Ghost After-Image
### Two versions of ghost after-image:
**1P ghost after image** - single ghost that follows original ghost
https://github.com/Clean-Code-Team/Invaders/assets/105896673/75151678-7962-40db-8ec6-56230328fea1
- **drawAfterEffect** method created
- original **drawGhost** method left as is, only added **drawAfterEffect** inside original code

**2P ghost after image** - many ghosts with different opacity that follow the original ghost 
https://github.com/Clean-Code-Team/Invaders/assets/105896673/a2c09091-353e-40fe-9ddd-cd440bf6e23b
- major change to **drawGhost2P** method *(maintains original animation logic)*

### CODE REVIEW

File: src/engine/DrawManager.java [hunk 1]
```diff
@@ -2612,70 +2612,150 @@ else if (0 < (255 - ((int)(System.currentTimeMillis() - ghostTImer) / 10)))
...
+//				this.drawEntity(SpriteType.Ghost, ghost1PostionX+tmp-2, ghost1PostionY+2, 2, 2, new Color(255,255,255,200));
+//				this.drawEntity(SpriteType.Ghost, ghost2PostionX+tmp-2, ghost2PostionY+2, 2, 2, new Color(255,255,255,200));
+//			}
+//			if (System.currentTimeMillis() - ghostTImer > 200){
+//				this.drawEntity(SpriteType.Ghost, ghost1PostionX+tmp-4, ghost1PostionY+4, 2, 2, new Color(255,255,255,170));
+//				this.drawEntity(SpriteType.Ghost, ghost2PostionX+tmp-4, ghost2PostionY+4, 2, 2, new Color(255,255,255,170));
+//			}
+//			if (System.currentTimeMillis() - ghostTImer > 300){
+//				this.drawEntity(SpriteType.Ghost, ghost1PostionX+tmp-6, ghost1PostionY+6, 2, 2, new Color(255,255,255,140));
+//				this.drawEntity(SpriteType.Ghost, ghost2PostionX+tmp-6, ghost2PostionY+6, 2, 2, new Color(255,255,255,140));
```
Comment (jeongwoo903) [inline src/engine/DrawManager.java]: [P2]

If it don't need, it would be better delete this comment.
It makes it harder to read.

**Review by jeongwoo903 — CHANGES_REQUESTED** [review]
[P1]

Reading the PR comments, it would be nice to see the ghost effect on 1p and 2p unified.
I would appreciate it if you could fix and merge this part.

File: src/engine/DrawManager.java (hunk 1 above)
Comment (jeongwoo903) [reply src/engine/DrawManager.java]: If it's not your code, feel free to leave it for now.

**Comment by alicek0** [comment]
> [P1]
> 
> Reading the PR comments, it would be nice to see the ghost effect on 1p and 2p unified. I would appreciate it if you could fix and merge this part.

@jeongwoo903 Thank you for your comment. Your feedback was considered and the ghost effect is now unified for 1P and 2P.

File: src/engine/DrawManager.java (hunk 1 above)
Comment (alicek0) [reply src/engine/DrawManager.java]: Thank you for your comment. I didn't write that, so we should leave it for now.

**Review by 258xsw — APPROVED** [review]
[P2]
I thought using 'if' was more appropriate in terms of time complexity than 'for' because the code was simple. What do you think?

**Review by jeongwoo903 — APPROVED** [review]

**Comment by alicek0** [comment]
> [P2]
> I thought using 'if' was more appropriate in terms of time complexity than 'for' because the code was simple. What do you think?

Thank you for your comment. The uses of 'if' and 'for' are independent of each other. The 'if' logic is the exact same as the original code, and the for loop is used for a different reason, which is to create duplicates of the ghost.

**Pull request merged**